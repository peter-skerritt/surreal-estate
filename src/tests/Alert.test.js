import React from "react";
import { render } from "@testing-library/react";
import Alert from "../components/Alert"

test('displays an error message', () => {
        const { getByText } = render(<Alert message="Error!" />);
      
        // eslint-disable-next-line testing-library/prefer-screen-queries
        expect(getByText(/Error/).textContent).toBe("Error!");
      });

      test("displays a success message", () => {
        const { getByText } = render(<Alert message="Success!!!!" success />);
      
        // eslint-disable-next-line testing-library/prefer-screen-queries
        expect(getByText(/Success/).textContent).toBe("Success!!!!");
      });

      test("does not render an error or a success message if message props is empty", () => {
        const { asFragment } = render(<Alert message="" />);
        const alert = asFragment();

        expect(alert).toMatchSnapshot();
      });