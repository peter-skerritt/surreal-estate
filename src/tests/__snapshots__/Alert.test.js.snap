// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not render an error or a success message if message props is empty 1`] = `<DocumentFragment />`;
